opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -677,16 +677,47 @@ namespace Opm {
             return pvSumLocal;
         }
 
+        double computeCnvErrorPv(const std::vector<Scalar>& B_avg, double dt)
+        {
+            double errorPV{};
+            const auto& ebosModel = ebosSimulator_.model();
+            const auto& ebosProblem = ebosSimulator_.problem();
+            const auto& ebosResid = ebosSimulator_.model().linearizer().residual();
+            const auto& gridView = ebosSimulator().gridView();
+            ElementContext elemCtx(ebosSimulator_);
+
+            for (const auto& elem: elements(gridView, Dune::Partitions::interiorBorder))
+            {
+                elemCtx.updatePrimaryStencil(elem);
+                elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+                const unsigned cell_idx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
+                const double pvValue = ebosProblem.referencePorosity(cell_idx, /*timeIdx=*/0) * ebosModel.dofTotalVolume( cell_idx );
+                const auto cellResidual = ebosResid[cell_idx];
```

Comment:
The `cellResidual` is subsequently being used as an array, meaning it's not **always** a scalar value.  What's the cost of taking a copy here versus using a reference?

Reply:
cellResidual has the size of number of equations. (i.e. typically 3) I agree that this should change to reference before merging this.

Reply:
Thanks for catching. Will fix in  a minute.